### PR TITLE
Manifest Function Updates

### PR DESCRIPTION
**Pull request by katy-sadowski**
_**Cleaned up some stuff to make the Capr agent work better, and got down a rabbit hole of fixes. Hope this all checks out. I didn't test it in the wild yet.**_

## Summary

Makes definition updates first-class across all manifest input routes, so edits propagate through the pipeline automatically and cohort IDs never churn.

### Capr
- New `updateCaprCohort()` / `updateCaprConceptSet()`: overwrite the registered JSON in place — same ID and file path, hash refreshed, derived dependents marked `stale`, no-op when the definition is unchanged.
- `addCaprCohort()` / `addCaprConceptSet()` gain `stopIfExists = FALSE` to upsert instead of erroring on an existing label (metadata refreshed; tags only replaced when supplied). `addCaprConceptSet()` now records `route = "capr"` for provenance parity with cohorts.

### Atlas
- `addAtlasCohort()` / `addAtlasConceptSet()` gain `stopIfExists = FALSE`: fetch the current ATLAS definition and update the registered entry in place.
- The load csv is now strictly a **one-time import**: `importAtlasCohorts()` / `importAtlasConceptSets()` fail fast (before importing anything) when rows are already registered by atlasId, or when labels collide.
- The importAtlas templates run `updateAtlasCohorts()` / `updateAtlasConceptSets()` unconditionally, **before** the import section — so every `main.R` run (via `sourceInputBuilderScripts()`) syncs registered definitions with ATLAS before the pipeline executes, and even a stale load csv cannot prevent the sync. The one-time import is skipped once the csv has been deleted (the documented post-import step), keeping `main.R` re-runs safe. Sync stays in input loading rather than pipeline execution so per-database runs never depend on ATLAS access and all databases execute identical frozen definitions.

### Derived cohorts
- All eight template builders (`buildUnionCohort`, `buildSubsetCohortTemporal`, `buildComplementCohort`, `buildCompositeCohort`, `buildDemographicCohort`, `buildOPriorT`, `buildTPriorO`, `buildCensorCohort`) plus `addDependentCustomCohort()` gain `stopIfExists = FALSE`: update parents, build parameters, and (for custom SQL) the file registration in place — SQL re-rendered, `depends_on`/`dependency_rule`/hash replaced, cohort marked `stale`, staleness cascaded to its own dependents.
- Dependency-cycle guard: a cohort cannot be updated to depend on itself or one of its own dependents.
- `dependentCohortIdList` entries now accept **vectors of cohort IDs** (rendered comma-separated by SqlRender, for `IN (@param)` clauses); all IDs flow into `depends_on`.

### Accidental-collision guards on all upserts
`stopIfExists = FALSE` under a typo'd label could silently replace an unrelated cohort's definition (same ID, changed meaning, no error). Accidental collisions almost always cross a provenance boundary the manifest records, so every upsert now verifies identity **before any mutation**:
- **Atlas** (cohorts + concept sets): the registered `atlasId` tag must match the atlasId passed; entries without an atlasId tag cannot be updated via the Atlas route. Registering the same atlasId under a second label is rejected as a duplicate import.
- **Capr**: `updateCaprCohort()` requires the registered `route = "capr"` tag (an ATLAS cohort can no longer be clobbered by a Capr update); `updateCaprConceptSet()` refuses ATLAS-registered concept sets.
- **Derived builders**: the existing cohort's type must match the type the builder creates (a union can only be updated by `buildUnionCohort()`, etc.).

Intentional cross-boundary changes remain possible as explicit two-steps (retag via `updateCohortTags()` / `updateConceptSetTags()`, or delete + rebuild), and every guard's error message says so.

### Deletion integrity
- `deleteCohort()` refuses to orphan derived dependents; new `cascade = TRUE` deletes the whole dependent subtree deepest-first (including DBMS cleanup with `dropFromDBMS = TRUE`).
- `syncManifest()` cascades soft-deletion to dependents when a parent's file has gone missing (new `cascade_deleted` action).
- `topological_sort()` reports dangling parent references explicitly instead of a misleading "possible circular dependency" error.

### Pipeline safety
- `sourceInputBuilderScripts()` no longer swallows builder-script errors: failures are aggregated across all scripts (so they can be fixed in one pass) and raised as a single abort — a failed input build can never be followed by `execStudyPipeline()` running on an incomplete or stale manifest. Previously the per-script `tryCatch` printed an alert and continued, and a scoping bug (missing `<<-` in the handlers) meant the returned `error_summary` was always empty.

## Bug fixes along the way
- Stale cohorts were unreachable by `generateCohorts()` — three `status = 'active'` filters dropped them from the dependency graph, the in-memory manifest, and dependency hashing, so the stale → regenerate → re-activate flow could never run.
- `checkAtlasConceptSets()` crashed on call (undefined `cm_atlas_subset` / `atlas_id`, nonexistent `filePath` column); `updateAtlasConceptSets()` targeted a nonexistent `timestamp` column.
- `updateAtlasCohorts()` never refreshed the in-memory manifest; fetch-failure handlers in all four Atlas check/update methods didn't skip failed iterations; early no-op exits fell through instead of returning.
- `deleteCohort()` fell through when the ID didn't exist and never refreshed the in-memory manifest.
- `addCaprConceptSet()` now raises a clean duplicate-label error instead of a raw SQLite unique-index failure.
- `sourceInputBuilderScripts()` error handlers assigned to handler-local copies of the errors list, so `error_summary` never reported anything.

## Tests
Expanded `test-CohortManifest-management.R` (80 passing), new `test-ConceptSetManifest-management.R` (33 passing), and new `test-sourceInputBuilderScripts.R` (6 passing), driving the flows end-to-end with fake ATLAS connections and Capr-generated fixtures: upserts keep IDs and sync hashes to disk, strict-import errors, collision-guard rejections (with nothing-changed assertions), stale cascade + generation reachability, delete guard/cascade, cycle rejection, the vector-ID round trip through SqlRender, and abort-on-builder-failure. Both importAtlas templates are verified to render through `glue::glue()` (the templates stay brace-free for this reason) and parse as valid R with sync ordered before import.

The `getManifestStatus` test is skipped pending the known `active_count` bug. Latest develop is merged in; pre-existing failures on develop are unchanged and untouched by this branch: 1 in query, 8 in builders, 7 in make, and 1 error in the new `test-CohortManifest-tags.R` (the test calls `renameTagKey(conceptSetIds = ...)`, an argument the function doesn't accept).

> **Note for local test runs:** the suite currently deletes the tracked fixture `tests/testthat/test_files/my_custom_cohort.sql` (pre-existing issue: `cm_test_add_sql_cohort` registers the fixture's real path and the `deleteCohort` test unlinks it). Restore with `git checkout -- tests/testthat/test_files/` afterwards; fixing the helper to copy fixtures into the temp repo is a good follow-up.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: inst/templates/loadScript_cohorts_importAtlas.R [hunk 1]
```diff
@@ -57,22 +57,45 @@ cohortManifest$setAtlasConnection(atlasConnection)
...
-# D. IMPORT COHORTS FROM ATLAS
+# D. SYNC REGISTERED ATLAS COHORTS
+# ================================================================================
+
+# Runs first, before any import: re-checks every registered ATLAS cohort
+# against ATLAS and updates changed definitions in place (same ID; derived
+# cohorts marked stale so the pipeline regenerates them). This is the step
+# that propagates ATLAS edits, and running it before the import means even a
+# stale load csv cannot prevent the manifest from syncing.
+cohortManifest$updateAtlasCohorts()
```
Comment (mdlavallee92) [inline inst/templates/loadScript_cohorts_importAtlas.R:68]: thx

File: R/execute.R [hunk 2]
```diff
@@ -1309,9 +1315,21 @@ sourceInputBuilderScripts <- function(
     cli::cli_alert_danger(
       "Error in sourceInputBuilderScripts(): {e$message}"
     )
-    errors[["critical"]] <- e$message
+    errors[["critical"]] <<- e$message
```
Comment (mdlavallee92) [inline R/execute.R:1318]: i dont like the `<<-` parent assignment. it can lead to unintended side effects but leave for now

File: R/cohort_builders.R [hunk 3]
```diff
@@ -159,6 +159,30 @@ topological_sort = function(graph) {
 
     # Verify all nodes were processed
     if (length(sorted_order) != length(graph)) {
+    unprocessed <- setdiff(as.integer(names(graph)), sorted_order)
+
+    # Distinguish dangling references (parent deleted/missing from the graph)
```
Comment (mdlavallee92) [inline R/cohort_builders.R:164]: i see so this cleans up deleted parents in the dependency graph

File: R/manifest_helpers.R [hunk 4]
```diff
@@ -1332,6 +1332,25 @@ cascadeStaleDownstream <- function(dbPath, cohort_ids) {
     }
   }
 
+  return(visited)
```
Comment (mdlavallee92) [inline R/manifest_helpers.R:1335]: nice so now stale scan goes up and down

**Review by mdlavallee92 — APPROVED** [review]
Thanks this is helpful! I will add this into v006 and rerelease. Please add a summary of the changes to the NEWS

File: R/execute.R (hunk 2 above)
Comment (katy-sadowski) [reply R/execute.R:1318]: Yeah I believe it's an antipattern for packages but apparently without it the errors weren't getting logged. I'll file an Issue to look into other ways to fix this.

File: R/cohort_builders.R (hunk 3 above)
Comment (katy-sadowski) [reply R/cohort_builders.R:164]: yeah Claude cleaned up some edge cases related to DAG getting broken when cohorts are deleted. it would throw an error if the manifest ended up in this state, but these changes should prevent it from happening at all.

that said i didn't test it in the wild yet 🙃
